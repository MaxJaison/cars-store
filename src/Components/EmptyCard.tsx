import { Card, CardContent } from "@material-ui/core";
import React from "react";
import "./CarsList.css";

const EmptyCard = () => {
  return (
    <>
      <Card>
        <CardContent className="content">
          <div className="grey-square"></div>
          <div>
            <div className="grey-line"></div>
            <div className="grey-line"></div>
            <div className="grey-line"></div>
          </div>
        </CardContent>
      </Card>
    </>
  );
};

export default EmptyCard;
