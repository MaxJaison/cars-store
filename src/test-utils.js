import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { applyMiddleware, createStore } from "redux";
import { Provider } from "react-redux";
import reducer from "./reducers";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from "@testing-library/react";
export { render };
