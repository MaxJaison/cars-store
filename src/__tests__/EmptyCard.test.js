import React from "react";
import { render } from "../test-utils";
import EmptyCard from "../components/EmptyCard";

test("renders EmptyCard", () => {
  const { container } = render(<EmptyCard />);
  expect(container).toMatchSnapshot();
});
