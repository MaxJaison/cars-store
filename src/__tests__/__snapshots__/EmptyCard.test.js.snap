// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EmptyCard 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root content"
    >
      <div
        class="grey-square"
      />
      <div>
        <div
          class="grey-line"
        />
        <div
          class="grey-line"
        />
        <div
          class="grey-line"
        />
      </div>
    </div>
  </div>
</div>
`;
